core/model/src/main/java/org/eclipse/rdf4j/model/util/Models.java

```diff
@@ -600,7 +600,7 @@ public static void convertRDFStarToReification(Model model, Consumer<Statement>
 	@Experimental
 	public static Model convertRDFStarToReification(ValueFactory vf, Model model) {
 		Model reificationModel = new LinkedHashModel();
-		convertRDFStarToReification(vf, model, reificationModel::add);
+		convertRDFStarToReification(vf, model, (Consumer<Statement>) reificationModel::add);
```

Comment:
This shouldn't be necessary, as far as I can tell the code is legal without the cast. Are you getting an actual error in IntelliJ, or is it just giving a warning of some sort?

I'm not really in favor of changing legal code just to make one particular editor behave (for one thing it becomes tricky to maintain), but if it's really unworkable, I'd rather go with a a proper lambda (`st -> reificationModel.add(st)`) instead of this explicit cast.

Reply:
<img width="1063" alt="Screenshot 2020-04-15 at 15 19 18" src="https://user-images.githubusercontent.com/797185/79341716-82ebc700-7f2c-11ea-8aa5-957f1437ac08.png">

Reply:
I prefer the explicit cast to creating a lambda.

What are you seeing for this code in your editor?

Reply:
> I prefer the explicit cast to creating a lambda.
 
Why? The lambda is arguably clearer to read, and definitely shorter.  I doubt the fact that it's a delegate has a serious impact (I would imagine the compiler optimizes that away). 

> What are you seeing for this code in your editor?

Nothing. Neither Eclipse nor the OpenJDK compiler report any issue with it (using OpenJDK 8 to build). I'm really surprised IntelliJ would complain about this. Is this a known issue? Perhaps it is having difficulty with the fact that `add` is overloaded.

Reply:
The lambda gets marked by the inspector as unnecessary while the casting gets ignored. 

It's most definitely at bug. Don't know if it's reported or not. It's very annoying though, since I can't do a full build from intellij anymore (which is needed to mitigate another bug I have).

Reply:
> The lambda gets marked by the inspector as unnecessary while the casting gets ignored.

That to me sounds more as if this inspector tool is configured a bit over-eager. But whatever, I don't like it, but if this is the only case then it's fairly trivial and not worth arguing over (at least not longer than we've alredy done ;)) - either way is legal Java.

Reply:
Reported the bug to the intellij people.